src/common/commonutils/FileUtils.c

```diff
@@ -617,41 +595,13 @@ static int CheckAccess(bool directory, const char* name, int desiredOwnerId, int
 
-                // S_IXOTH (00001): Execute/search permission, others
-                // S_IWOTH (00002): Write permission, others
-                // S_IROTH (00004): Read permission, others
-                // S_IRWXO (00007): Read, write, execute/search by others
-                // S_IXGRP (00010): Execute/search permission, group
-                // S_IWGRP (00020): Write permission, group
-                // S_IRGRP (00040): Read permission, group
-                // S_IRWXG (00070): Read, write, execute/search by group
-                // S_IXUSR (00100): Execute/search permission, owner
-                // S_IWUSR (00200): Write permission, owner
-                // S_IRUSR (00400): Read permission, owner
-                // S_IRWXU (00700): Read, write, execute/search by owner
-                // S_ISVTX (01000): On directories, restricted deletion flag
-                // S_ISGID (02000): Set-group-ID on execution
-                // S_ISUID (04000): Set-user-ID on execution
-
-                currentMode = DecimalToOctal(statStruct.st_mode & 07777);
-                desiredMode = desiredAccess;
-
-                if (((desiredMode & S_IRWXU) && ((desiredMode & S_IRWXU) != (currentMode & S_IRWXU))) ||
-                    ((desiredMode & S_IRWXG) && ((desiredMode & S_IRWXG) != (currentMode & S_IRWXG))) ||
-                    ((desiredMode & S_IRWXO) && ((desiredMode & S_IRWXO) != (currentMode & S_IRWXO))) ||
-                    ((desiredMode & S_IRUSR) && ((desiredMode & S_IRUSR) != (currentMode & S_IRUSR))) ||
-                    ((desiredMode & S_IRGRP) && ((desiredMode & S_IRGRP) != (currentMode & S_IRGRP))) ||
-                    ((desiredMode & S_IROTH) && ((desiredMode & S_IROTH) != (currentMode & S_IROTH))) ||
-                    ((desiredMode & S_IWUSR) && ((desiredMode & S_IWUSR) != (currentMode & S_IWUSR))) ||
-                    ((desiredMode & S_IWGRP) && ((desiredMode & S_IWGRP) != (currentMode & S_IWGRP))) ||
-                    ((desiredMode & S_IWOTH) && ((desiredMode & S_IWOTH) != (currentMode & S_IWOTH))) ||
-                    ((desiredMode & S_IXUSR) && ((desiredMode & S_IXUSR) != (currentMode & S_IXUSR))) ||
-                    ((desiredMode & S_IXGRP) && ((desiredMode & S_IXGRP) != (currentMode & S_IXGRP))) ||
-                    ((desiredMode & S_IXOTH) && ((desiredMode & S_IXOTH) != (currentMode & S_IXOTH))) ||
-                    ((desiredMode & S_ISUID) && ((desiredMode & S_ISUID) != (currentMode & S_ISUID))) ||
-                    ((desiredMode & S_ISGID) && ((desiredMode & S_ISGID) != (currentMode & S_ISGID))) ||
-                    (directory && (desiredMode & S_ISVTX) && ((desiredMode & S_ISVTX) != (currentMode & S_ISVTX))) ||
-                    (currentMode > desiredMode))
+                currentMode = statStruct.st_mode & 07777;
+                desiredMode = desiredAccess & 07777;
+
+                if (!directory) {
+                    desiredMode &= ~S_ISVTX;
+                }
```

Comment:
Minor reformatting suggested:

```
if (!directory) 
{
    desiredMode &= ~S_ISVTX;
}
// <- separator line here
if (currentMode != desiredMode)
```

Also, would be good to add in one line comment meaning of S_ISVTX, basically keeping this line:

```
// S_ISVTX (01000): On directories, restricted deletion flag
```

(this may be familiar to Linux experts like you, but not to everyone else who may need to understand this code)

Reply:
Fixed.

Reply:
Thanks! Sorry if I wasn't clear enough. I took the liberty to merge dev into your branch and suggest in a new commit. Rest is LGTM to me. I will re-run the CI jobs that failed and ask for a 2nd CR from team. If all will go well, I'll merge this in today, TBD.